golem-worker-service-base/src/worker_bridge_execution/content_type_mapper.rs

```diff
@@ -0,0 +1,223 @@
+                }),
+                // Fail if it is not a binary stream
+                other => Err(ContentTypeMapError::expect_only_binary_stream(other)),
+            },
+            // Str is equivalent to binary stream, therefore proceeds with responding as a binary stream
+            TypeAnnotatedValue::Str(str) => {
+                Ok(Body::from_bytes(bytes::Bytes::from(str.to_string()))
+                    .with_content_type(non_json_content_type.to_string()))
+            }
+            other => Err(ContentTypeMapError::expect_only_binary_stream(
+                &AnalysedType::from(other),
+            )),
+        }
+    }
+
+    fn get_byte_stream(values: &Vec<TypeAnnotatedValue>) -> Result<Vec<u8>, ContentTypeMapError> {
+        let bytes = values
+            .into_iter()
+            .map(|v| match v {
+                TypeAnnotatedValue::U8(u8) => Ok(*u8),
+                _ => Err(ContentTypeMapError::internal(
+                    "The analysed type is a binary stream however unable to fetch vec<u8>",
+                )),
+            })
+            .collect::<Result<Vec<_>, ContentTypeMapError>>()?;
+
+        Ok(bytes)
+    }
+
+    fn get_byte_stream_body(
+        values: &Vec<TypeAnnotatedValue>,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        let bytes = get_byte_stream(values)?;
+        Ok(Body::from_bytes(bytes::Bytes::from(bytes))
+            .with_content_type(ContentType::octet_stream().to_string()))
+    }
+
+    fn get_json(
+        type_annotated_value: &TypeAnnotatedValue,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        let json = get_json_from_typed_value(type_annotated_value);
+        Body::from_json(json)
+            .map(|body| body.with_content_type(ContentType::json().to_string()))
+            .map_err(|_| ContentTypeMapError::internal("Failed to convert to json body"))
+    }
+
+    // If the request Content-Type is application/json, then we can allow ANY component model type of value as the response.
+    // However, only Rib return values which are not equivalent to a binary stream are automatically converted into JSON (jsonification).
+    fn get_json_or_binary_stream(
+        type_annotated_value: &TypeAnnotatedValue,
+    ) -> Result<WithContentType<Body>, ContentTypeMapError> {
+        match type_annotated_value {
+            TypeAnnotatedValue::List { typ, values } => match typ {
+                // It is already a binary stream and we keep it as is
+                AnalysedType::U8 => get_byte_stream_body(values),
+                // Convert to Json otherwise
+                _ => get_json(type_annotated_value),
+            },
+            // A string is considered as a binary stream
+            TypeAnnotatedValue::Str(str) => Ok(get_text_body(str)),
+            _ => get_json(type_annotated_value),
+        }
+    }
+
+    fn get_null() -> Result<WithContentType<Body>, ContentTypeMapError> {
```

Comment:
i think for clarity recommend to rename it 
get_null -> get_json_null

Reply:
Sure @justcoon